### PR TITLE
upload file implementation lightning ue

### PR DESCRIPTION
**Pull request by JuanCarlosMartinez4**

### CODE REVIEW

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 1]
```diff
@@ -71,22 +75,26 @@ public static AbstractCampaignListPage getCampaignsPage() {
...
      *
-     * @return OpportunityPage instance.
+     * @return CampaignsPage instance.
      */
     public static AbstractOpportunityPage getOpportunityPage() {
-        if (userExperience.equals("Classic")) {
+        if (UserExperience.USER_EXPERIENCE_CLASSIC.get().equals(userExperience)) {
             return new OpportunityClassicPage();
+        } else if (UserExperience.USER_EXPERIENCE_LIGHTNING.get().equals(userExperience)) {
+            return new OpportunityLightningPage();
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java]: review documentation please. It returns an opportunity

File: src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    @FindBy(xpath = "//a[@title='Upload Files']")
+    private WebElement uploadFilesButton;
+
+    /**
+     * Gets a web element by xpath.
+     * @param xpathValue string.
+     * @param textToConcat value.
+     * @return a web element.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java]: crl + alt + L please

File: src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        clickOnUploadFilesButton();
+        UploadFileRobot.uploadFile(fileAbsolutePath);
+        closeUploadFilesPopUp();
+    }
+
+    /**
+     * Gets uploaded file name.
+     * @param fileName param.
+     * @return uploaded file name.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsLightningPage.java]: crl + alt + L please

File: src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsPageAbstract.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @author Juan Martinez.
+ * @version 1.0 21 March 2020.
+ */
+public abstract class NotesAndAttachmentsPageAbstract extends AbstractBasePage {
+
+    /**
+     * Clicks on upload files.
+     * @param filePath value.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/notesattachments/NotesAndAttachmentsPageAbstract.java]: crl + alt + L please

File: src/main/java/salesforce/utils/UploadFileRobot.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public static void setClipboardData(final String fileLocation) {
+        StringSelection stringSelection = new StringSelection(fileLocation);
+        Toolkit.getDefaultToolkit().getSystemClipboard().setContents(stringSelection, null);
+    }
+
+    /**
+     * Allows to upload file.
+     * @param fileLocation path.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/utils/UploadFileRobot.java]: crl + alt + L please

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 6]
```diff
@@ -66,27 +70,31 @@ public static AbstractCampaignListPage getCampaignsPage() {
...
      */
     public static AbstractOpportunitiesPage getOpportunitiesPage() {
-        if (userExperience.equals("Classic")) {
+        if (UserExperience.USER_EXPERIENCE_CLASSIC.get().equals(userExperience)) {
             return new OpportunitiesClassicPage();
+        } else if (UserExperience.USER_EXPERIENCE_LIGHTNING.get().equals(userExperience)) {
+            return new OpportunitiesLightningPage();
         }
-        return new OpportunitiesLightningPage();
+        return null;
```
Comment (Enrique-C) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java]: All of the would be e.g.: 
       if (userExperience.equals("Classic")) {
            return new NewCampaignClassicPage();
        }
        return new NewCampaignLightningPopUp();

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 7]
```diff
@@ -66,27 +70,31 @@ public static AbstractCampaignListPage getCampaignsPage() {
...
      *
-     * @return OpportunityPage instance.
+     * @return CampaignsPage instance.
      */
     public static AbstractOpportunityPage getOpportunityPage() {
-        if (userExperience.equals("Classic")) {
+        if (UserExperience.USER_EXPERIENCE_CLASSIC.get().equals(userExperience)) {
             return new OpportunityClassicPage();
+        } else if (UserExperience.USER_EXPERIENCE_LIGHTNING.get().equals(userExperience)) {
+            return new OpportunityLightningPage();
```
Comment (Enrique-C) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java]: The same here

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 8]
```diff
@@ -124,4 +132,18 @@ public static AbstractCampaignPage getCampaignPage() {
...
+     *
+     * @return NotesAndAttachmentsPage instance.
+     */
+    public static NotesAndAttachmentsPageAbstract getNotesAndAttachments() {
+        if (UserExperience.USER_EXPERIENCE_CLASSIC.get().equals(userExperience)) {
+            return new NotesAndAttachmentsClassicPage();
+        } else if (UserExperience.USER_EXPERIENCE_LIGHTNING.get().equals(userExperience)) {
+            return new NotesAndAttachmentsLightningPage();
+        }
+        return null;
```
Comment (Enrique-C) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java]: The same here

**Review by Enrique-C — COMMENTED** [review]
src/test/resources/filestoupload/cucumber.png What does it?

File: src/main/java/salesforce/ui/pages/AppPageFactory.java (hunk 6 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/AppPageFactory.java]: that case is not completly valid
Although that is harcoded  **equals("Classic")) {**

Comment (Enrique-C) [reply src/main/java/salesforce/ui/pages/AppPageFactory.java]: You are true, it would be a CONSTANT

**Review by Enrique-C — APPROVED** [review]

**Pull request merged**